airbyte-commons/src/main/java/io/airbyte/commons/io/LineGobbler.java

```diff
@@ -30,13 +30,18 @@
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.function.Consumer;
+import java.util.function.Function;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class LineGobbler implements VoidCallable {
 
   private final static Logger LOGGER = LoggerFactory.getLogger(LineGobbler.class);
 
+  public static void gobble(final InputStream is, final Logger logger, final Function<Logger, Consumer<String>> consumerFactory) {
```

Comment:
It doesn't seem necessary to add this function. We are trying to generalize a behavior that is integration specific

Reply:
We could also reuse the previous consumer pattern without the need for that factory since you're passing the logger.

Reply:
I was using the factory method to ensure that the default log consumer was a method on the same logger, but I guess that's not necessary. I'll switch.